File: create_dirs.py
#!/usr/bin/env python3
import os
from pathlib import Path

base_path = Path(r"c:\Users\tomas\Documents\vscode\RunthisRepo\run-this-repo")

dirs_to_create = [
    "lib/supabase",
    "app/login",
    "app/dashboard/analyses/[id]",
    "app/api/auth/logout",
    "app/api/analyses/save"
]

for dir_path in dirs_to_create:
    full_path = base_path / dir_path
    full_path.mkdir(parents=True, exist_ok=True)
    print(f"Created: {full_path}")

print("All directories created successfully!")

File: create-dirs.js
const fs = require('fs');
const path = require('path');

const basePath = 'c:\\Users\\tomas\\Documents\\vscode\\RunthisRepo\\run-this-repo';

const dirs = [
  path.join(basePath, 'lib', 'supabase'),
  path.join(basePath, 'app', 'login'),
  path.join(basePath, 'app', 'dashboard', 'analyses', '[id]'),
  path.join(basePath, 'app', 'api', 'auth', 'logout'),
  path.join(basePath, 'app', 'api', 'analyses', 'save')
];

let output = 'Creating directories...\n\n';

// Create directories
dirs.forEach(dir => {
  try {
    if (!fs.existsSync(dir)) {
      fs.mkdirSync(dir, { recursive: true });
      output += `✓ Created: ${dir}\n`;
    } else {
      output += `✓ Already exists: ${dir}\n`;
    }
  } catch (err) {
    output += `✗ Error creating ${dir}: ${err.message}\n`;
  }
});

output += '\nVerifying directories...\n\n';

// Verify directories exist
dirs.forEach(dir => {
  try {
    if (fs.existsSync(dir)) {
      const stats = fs.statSync(dir);
      output += `✓ Verified: ${dir}\n`;
    } else {
      output += `✗ Not found: ${dir}\n`;
    }
  } catch (err) {
    output += `✗ Error verifying ${dir}: ${err.message}\n`;
  }
});

console.log(output);

// Also save to file for verification
fs.writeFileSync(path.join(basePath, '.dir-creation-report.txt'), output);
